### PR TITLE
docs(core): restructure plugin docs

### PR DESCRIPTION
**Pull request by isaacplmann**
Restructures the create your own plugin section

### CODE REVIEW

**Comment by ZackDeRose** [comment]
Things I've noted that don't look like we can tie directly to a change:

+ I think rephrasing our top-level descriptions of generators and executors would be wise. I think the following will resonate:
  + "The goal of Nx’s Generator API is to simplify any code-generation script down to a simple function, and to provide utility functions (via `@nrwl/devkit`) to make the more burdensome things about code-generation scripts more manageable and tolerable."
  + "Our goal in Nx’s task-running API is to simplify any task down to a single function: an executor. When we use Nx to run a task (like with the command: `npx nx build my-app`) Nx will simply determine the executor function attached to `my-app`'s build target, and call that executor function."

^ Emphasis here on Nx simplifying these efforts to a single function; and removing burdensome, low-impact/high-cost tasks from engineering.

I think this would do well to replace some of the stuff at the top of `docs/shared/packages/nx-plugin/nx-plugin.md`!

The page here: https://nx.dev/plugin-features would be nice to reflect most of what's in `docs/shared/packages/nx-plugin/nx-plugin.md`! Right now it feels especially bare.

Screen Shot 2023-01-13 at 10 10 32 AM

File: docs/shared/recipes/plugins/create-preset.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+## Preset
+
+A Preset is a customization option which you provide when creating a new workspace. TS, Node, React are some internal presets that Nx provides by default.
```
Comment (ZackDeRose) [inline docs/shared/recipes/plugins/create-preset.md:3]:
```suggestion
When you create a fresh nx workspace, you run the command: `npx create-nx-workspace` (w/link!).

This command accepts a `--preset` option, for example: `npx create-nx-workspace --preset=react-standalone`.

This preset option is pointing to a special generator function (remember, a generator is a function that simplifies an entire code generation script into a single function) that Nx will call when this `npx create-nx-workspace` command is run, that will generate your initial workspace.
```

File: docs/shared/recipes/plugins/create-preset.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+{% youtube
+src="https://www.youtube.com/embed/yGUrF0-uqaU"
+title="Develop a Nx Preset for your Nx Plugin"
+width="100%" /%}
+
+### Custom Preset
+
+At its core a preset is a generator, which we can create inside of a plugin.
```
Comment (ZackDeRose) [inline docs/shared/recipes/plugins/create-preset.md:12]: Add here:

"All first-party Nx presets are built into nx itself, but you can [Create Your Own Plugin](link) and create a generator with the magic name: "preset". Once you've [Published Your Plugin](link) on npm, you can now run. the `create-nx-workspace` command with the preset option set to the name of your published package.

For example, take

```terminal
npx create-nx-workspace --preset=qwik-nx
```

This command will create a new qwik preset based on [the published npm package: "qwik-nx](https://www.npmjs.com/package/qwik-nx). If we check this package's source code, we can see that it has a [generator named "preset"](https://github.com/qwikifiers/qwik-nx/tree/main/packages/qwik-nx/src/generators/preset)

File: docs/shared/recipes/plugins/migration-generators.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
+# Migration Generators
+
+When your plugin is being used in other repos, it is helpful to provide migration generators to automatically update configuration files when your plugin makes a breaking change.
+
+A migration generator is a normal generator that is triggered when a developer runs the `nx migrate` command.
```
Comment (ZackDeRose) [inline docs/shared/recipes/plugins/migration-generators.md:5]: 👍 🔥

File: docs/shared/recipes/plugins/share-your-plugin.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+- Use the following commit message template: `chore(core): nx plugin submission [PLUGIN_NAME]`
+- push your changes, and run `yarn submit-plugin`
+
+> The `yarn submit-plugin` command automatically opens the Github pull request process with the correct template.
+
+We will then verify the plugin, offer suggestions or merge the pull request!
+
+## Write Migrations
+
+Once other repos are using your plugin, it would help them if you write migrations to automatically update their configuration files whenever you make breaking changes. Read the [migration generators guide](/recipes/advanced-plugins/migration-generators) to find out how.
```
Comment (ZackDeRose) [inline docs/shared/recipes/plugins/share-your-plugin.md:31]: 👍 🔥

File: nx-dev/ui-markdoc/src/lib/tags/personas.component.tsx [hunk 5]
```diff
@@ -134,6 +141,31 @@ const typeMap: Record<
...
+          d="M11.987 14.138l-3.132 4.923-5.193-8.427-.012 8.822H0V4.544h3.691l5.247 8.833.005-3.998 3.044 4.759zm.601-5.761c.024-.048 0-3.784.008-3.833h-3.65c.002.059-.005 3.776-.003 3.833h3.645zm5.634 4.134a2.061 2.061 0 0 0-1.969 1.336 1.963 1.963 0 0 1 2.343-.739c.396.161.917.422 1.33.283a2.1 2.1 0 0 0-1.704-.88zm3.39 1.061c-.375-.13-.8-.277-1.109-.681-.06-.08-.116-.17-.176-.265a2.143 2.143 0 0 0-.533-.642c-.294-.216-.68-.322-1.18-.322a2.482 2.482 0 0 0-2.294 1.536 2.325 2.325 0 0 1 4.002.388.75.75 0 0 0 .836.334c.493-.105.46.36 1.203.518v-.133c-.003-.446-.246-.55-.75-.733zm2.024 1.266a.723.723 0 0 0 .347-.638c-.01-2.957-2.41-5.487-5.37-5.487a5.364 5.364 0 0 0-4.487 2.418c-.01-.026-1.522-2.39-1.538-2.418H8.943l3.463 5.423-3.379 5.32h3.54l1.54-2.366 1.568 2.366h3.541l-3.21-5.052a.7.7 0 0 1-.084-.32 2.69 2.69 0 0 1 2.69-2.691h.001c1.488 0 1.736.89 2.057 1.308.634.826 1.9.464 1.9 1.541a.707.707 0 0 0 1.066.596zm.35.133c-.173.372-.56.338-.755.639-.176.271.114.412.114.412s.337.156.538-.311c.104-.231.14-.488.103-.74z"
+          fill="white"
+        />
+      </svg>
+    ),
+  },
+  browse: {
+    image: <DocumentMagnifyingGlassIcon role="img" className="h-6 w-6" />,
+  },
```
Comment (ZackDeRose) [inline nx-dev/ui-markdoc/src/lib/tags/personas.component.tsx:168]: the new personas looks epic!!! (see other comment that I think we should move it to the top level docs!!

**Comment by isaacplmann** [comment]
I've updated the generator and executor description at the top of the nx-plugin doc, but I didn't use the exact wording you suggested.

The /plugin-features/ page is a landing page to just redirect you to other pages.  None of the other landing pages have any content on them.

**Review by ZackDeRose — APPROVED** [review]
This looks good to me! Thanks for adding those in. I still would really like a 4-square personas thing for the plugins overview page (the four categories being maybe Generators | Executors | Project Graph | Create Your Own) but maybe we can discuss it later.

**Pull request merged**